src/stories/containers/Finances/utils/utils.ts

```diff
@@ -716,3 +717,51 @@ export const sortDataByElementCount = (data: MockData) => {
 
   return sortedData;
 };
+
+export const showOthersFooterRow = (data: MockData) => {
+  let totalRows = 0;
+  Object.keys(data).forEach((key) => {
+    totalRows += data[key].length;
+  });
+  console.log({ totalRows });
+  if (totalRows > 16) return true;
+  return false;
+};
+
+// Get first element of each table that always have to appear
+const getFirstElementEachTable = (data: MockData) => {
+  const moment: MockData = {};
+  Object.keys(data).forEach((element) => {
+    moment[element] = data[element].slice(0, 1);
+  });
+  return moment;
+};
+
+export const showOnlySixteenRowsWithOthers = (data: MockData) => {
+  const result = getFirstElementEachTable(data);
+
+  const maxRows = NUMBER_ROWS_FINANCES_TABLE;
+  let totalRowsPerTable = 0;
+  const firstElements: Record<string, QuarterlyBudget[]> = {};
+  const totalRows = Object.keys(data).reduce((acc, key) => acc + data[key].length, 0);
+  // Si la cantidad total de filas es menor o igual a 16, no necesitas hacer cambios
+  if (totalRows < maxRows) {
+    return data;
+  }
+
+  // Save the first element of each table
+  Object.keys(data).forEach((key) => {
+    firstElements[key] = [data[key][0]];
```

Comment:
are we 100% sure that `data[key]` will always be an array with at least 1 element? If not, we should prepare to receive empty arrays here, if that happens we could have an error here

Reply:
You are right